### DISCUSSION TITLE
nestjs-pothos: Use pothos as GraphQL schema builder in Nest.js application

### DISCUSSION THREAD

**Discussion by iamchanii** [hayes/pothos#742] [Show and tell]
I just want to show you and get some feedbacks of any kind. recently I'm working on to try to use Pothos with Nestjs application. is anyone interested of it? I plan to continue to develop by receiving feedback and using it in my company's work.

https://github.com/smatch-corp/nestjs-pothos

**Comment by hayes** [comment]
Nice!  I don't know a lot about nest, and haven't used DI based frameworks much myself, but this looks pretty cool!  I've definitely wondered if there was a good way to integrate Pothos and next, since they have their own GraphQL integration already, but would be cool if this end up working well for you!

I've been meaning to create a place in the Pothos docs to share cool 3rd party libraries, examples, and tools, so definitely keep me posted on how this goes!

**Reply by iamchanii** [reply]
I'd glad to hear some feedbacks from you! I've done implement main concept of it and integrate with Nest's GraphQLModule. if you want to check:

https://github.com/smatch-corp/nestjs-pothos/tree/main/examples/with-express-apollo

From I decided to use Pothos with Yoga, I didn't consider to use with GraphQLModule. but maybe someone still needs it. So I wrote a few more lines of code, but I think that's the only way to focus build a GraphQL schema. (and I believe it is same with pothos's way)

BTW, I started my job work with nestjs-pothos. I'll let you know again if there's anything special.

**Reply by hayes** [reply]
I looked at this briefly, and it looks pretty cool!  Unfortunately I don't have the time to do a deep dive on this (would require learning a lot about nest, and how it does dependency injection) because I have a lot of other Pothos projects in flight, and not nearly enough time to work on all of them,  so I don't have any in depth feedback at the moment, but definitely interested to hear how this ends up working for you!

**Comment by macrozone** [comment]
@iamchanii I really appreciate the work you did! It looks very promising

I think nestjs has its merrits, like DI. But its default graphql implementation is far from ideal. Its not 100% sound and typesafe unlike pothos which has a very strong type safety. (also personally i dislike decorators, but thats a matter of taste i guess). So I think pothos is a nice replacement.

I really hope your repository gets more traction from other devs so we can confidently use it in production soon.

**Reply by iamchanii** [reply]
Thanks! your feedback makes me happy :)  I haven't been able to do a lot of work lately because I've been busy, but I hope so as possible.

**Comment by printmart** [comment]
https://github.com/smatch-corp/nestjs-pothos
This one is very well. It should be added into the pothos doc. Thank you so much @iamchanii

**Reply by hayes** [reply]
added a new section to the docs for 3rd party libraries with a link to this
